test/lint/lint-circular-dependencies.sh

```diff
@@ -15,6 +15,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "index/base -> validation -> index/blockfilterindex -> index/base"
     "index/coinstatsindex -> node/coinstats -> index/coinstatsindex"
     "policy/fees -> txmempool -> policy/fees"
+    "policy/rbf -> txmempool -> validation -> policy/rbf"
```

Comment:
Concept ACK, but I think we should avoid adding a circular dependency here.

Reply:
Yep! I've rebased #22677 on top of this PR so it's more clear that it would fix this problem.

Reply:
Okie done